### PR TITLE
Upgrade to 2022

### PR DESCRIPTION
**Pull request by Piphi5**
~~TODO:~~
- [ ] Add Romi Gyro
- [ ] Uncomment everything once all dependencies are installed

~~Remaining Vendor Builds:~~
- [ ] CTRE
- [ ] REV
- [ ] Venom
- [ ] NavX (Mainly need linux + mac binaries)

Edit: This PR will now serve to only upgrade robot code to 2022 with limited vendor support for beta testing. Vendors will be added in subsequent PRs.

### CODE REVIEW

File: sysid-projects/mechanism/src/main/cpp/Robot.cpp [hunk 1]
```diff
@@ -43,22 +41,20 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
...
-    wpi::outs().flush();
+    fmt::print("{}\n", "Initializing Motors");
     for (size_t i = 0; i < ports.size(); i++) {
       AddMotorController(ports[i], controllerNames[i], motorsInverted[i],
                          &m_controllers);
     }
 
-    wpi::outs() << "Initializing encoder\n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Initializing encoder");
```
Comment (calcmogul) [inline sysid-projects/mechanism/src/main/cpp/Robot.cpp:50]:
```suggestion
    fmt::print("Initializing encoder\n");
```

File: sysid-library/src/main/cpp/logging/SysIdLogger.cpp [hunk 2]
```diff
@@ -59,8 +58,9 @@ void SysIdLogger::UpdateThreadPriority() {
 }
 
 SysIdLogger::SysIdLogger() {
+  fmt::print("{}\n", "Initializing logger");
```
Comment (calcmogul) [inline sysid-library/src/main/cpp/logging/SysIdLogger.cpp:61]:
```suggestion
  fmt::print("Initializing logger\n");
```

File: sysid-projects/analysis-test/src/main/cpp/Drivetrain.cpp [hunk 3]
```diff
@@ -47,6 +48,8 @@ void Drivetrain::SimulationPeriodic() {
                                       frc::RobotController::GetInputVoltage());
   m_drivetrainSimulator.Update(5_ms);
 
+  fmt::print("{}\n", m_drivetrainSimulator.GetLeftPosition().to<double>());
```
Comment (calcmogul) [inline sysid-projects/analysis-test/src/main/cpp/Drivetrain.cpp:51]:
```suggestion
  fmt::print("{}\n", m_drivetrainSimulator.GetLeftPosition().value());
```

File: sysid-projects/analysis-test/src/main/cpp/Robot.cpp [hunk 4]
```diff
@@ -5,13 +5,12 @@
...
 
-#include <frc/SlewRateLimiter.h>
 #include <frc/TimedRobot.h>
 #include <frc/XboxController.h>
+#include <frc/filter/SlewRateLimiter.h>
 #include <frc/simulation/DriverStationSim.h>
 #include <frc/smartdashboard/SmartDashboard.h>
 #include <units/voltage.h>
-#include <wpi/StringRef.h>
 #include <wpi/raw_ostream.h>
```
Comment (calcmogul) [inline sysid-projects/analysis-test/src/main/cpp/Robot.cpp:14]: wpi/raw_ostream.h is unused.

File: sysid-projects/drive/src/main/cpp/Robot.cpp [hunk 5]
```diff
@@ -25,8 +26,8 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
   m_json = GetConfigJson();
 
   try {
-    wpi::outs() << "Read JSON\n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Read JSON");
```
Comment (calcmogul) [inline sysid-projects/drive/src/main/cpp/Robot.cpp:29]:
```suggestion
    fmt::print("Read JSON\n");
```

File: sysid-projects/drive/src/main/cpp/Robot.cpp [hunk 6]
```diff
@@ -60,28 +61,25 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
 
     int period = m_json.at("velocity measurement period").get<int>();
 
-    wpi::outs() << "Setup motors\n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Setup motors");
```
Comment (calcmogul) [inline sysid-projects/drive/src/main/cpp/Robot.cpp:64]:
```suggestion
    fmt::print("Setup motors\n");
```

File: sysid-projects/drive/src/main/cpp/Robot.cpp [hunk 7]
```diff
@@ -60,28 +61,25 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
...
     for (size_t i = 0; i < leftPorts.size(); i++) {
       AddMotorController(leftPorts[i], controllerNames[i],
                          leftMotorsInverted[i], &m_leftControllers);
       AddMotorController(rightPorts[i], controllerNames[i],
                          rightMotorsInverted[i], &m_rightControllers);
     }
 
-    wpi::outs() << "Setup encoders\n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Setup encoders");
```
Comment (calcmogul) [inline sysid-projects/drive/src/main/cpp/Robot.cpp:72]:
```suggestion
    fmt::print("Setup encoders\n");
```

File: sysid-projects/drive/src/main/cpp/Robot.cpp [hunk 8]
```diff
@@ -60,28 +61,25 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
...
                   m_leftEncoder, m_leftPosition, m_leftRate);
     SetupEncoders(encoderType, isEncoding, period, cpr * gearing, numSamples,
                   controllerNames[0], m_rightControllers.at(0).get(),
-                  rightEncoderInverted, rightEncoderPorts, m_rightCancoder,
+                  rightEncoderInverted, rightEncoderPorts, // m_rightCancoder,
                   m_rightEncoder, m_rightPosition, m_rightRate);
 
-    wpi::outs() << "setup gyro\n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Setup gyro");
```
Comment (calcmogul) [inline sysid-projects/drive/src/main/cpp/Robot.cpp:82]:
```suggestion
    fmt::print("Setup gyro\n");
```

File: sysid-projects/drive/src/main/cpp/Robot.cpp [hunk 9]
```diff
@@ -257,7 +256,7 @@ void Robot::TeleopPeriodic() {}
 void Robot::DisabledInit() {
   SetMotorControllers(0_V, m_leftControllers);
   SetMotorControllers(0_V, m_rightControllers);
-  wpi::outs() << "Robot disabled\n";
+  fmt::print("{}\n", "Robot disabled");
```
Comment (calcmogul) [inline sysid-projects/drive/src/main/cpp/Robot.cpp:259]:
```suggestion
  fmt::print("Robot disabled\n");
```

File: sysid-projects/mechanism/src/main/cpp/Robot.cpp [hunk 10]
```diff
@@ -8,20 +8,18 @@
...
 #include "generation/SysIdSetup.h"
 
 Robot::Robot() : frc::TimedRobot(5_ms) {
-  m_json = GetConfigJson();
-
   try {
-    wpi::outs() << "reading json \n";
-    wpi::outs().flush();
+    m_json = GetConfigJson();
+    fmt::print("{}\n", "Reading JSON");
```
Comment (calcmogul) [inline sysid-projects/mechanism/src/main/cpp/Robot.cpp:22]:
```suggestion
    fmt::print("Reading JSON\n");
```

File: sysid-projects/mechanism/src/main/cpp/Robot.cpp [hunk 11]
```diff
@@ -43,22 +41,20 @@ Robot::Robot() : frc::TimedRobot(5_ms) {
 
     int period = m_json.at("velocity measurement period").get<int>();
 
-    wpi::outs() << "Initializing motors \n";
-    wpi::outs().flush();
+    fmt::print("{}\n", "Initializing Motors");
```
Comment (calcmogul) [inline sysid-projects/mechanism/src/main/cpp/Robot.cpp:44]:
```suggestion
    fmt::print("Initializing Motors\n");
```

File: sysid-projects/mechanism/src/main/cpp/Robot.cpp [hunk 12]
```diff
@@ -119,7 +114,7 @@ void Robot::TeleopPeriodic() {}
 
 void Robot::DisabledInit() {
   SetMotorControllers(0_V, m_controllers);
-  wpi::outs() << "Robot disabled\n";
+  fmt::print("{}\n", "Robot Disabled");
```
Comment (calcmogul) [inline sysid-projects/mechanism/src/main/cpp/Robot.cpp:117]:
```suggestion
  fmt::print("Robot Disabled\n");
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: sysid-projects/analysis-test/src/main/include/Drivetrain.h [hunk 13]
```diff
@@ -4,14 +4,16 @@
...
 #pragma once
 
+#include <fmt/format.h>
 #include <frc/AnalogGyro.h>
 #include <frc/Encoder.h>
-#include <frc/PWMVictorSPX.h>
-#include <frc/SpeedControllerGroup.h>
 #include <frc/controller/PIDController.h>
 #include <frc/controller/SimpleMotorFeedforward.h>
+#include <frc/fmt/Eigen.h>
```
Comment (calcmogul) [inline sysid-projects/analysis-test/src/main/include/Drivetrain.h:12]: What uses `frc/fmt/Eigen.h`?

Comment (Piphi5) [reply sysid-projects/analysis-test/src/main/include/Drivetrain.h:12]: I think that was from us troubleshooting the angular drivetrain test

**Review by calcmogul — DISMISSED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**